### ISSUE TITLE
Insufficient training results on the attached data

### ISSUE THREAD

**Issue by andreqwert** [aman5319/Multi-Label#1]
Hello,
Thanks for sharing your script!
I tried to launch your code and faced a problem on the first training step (section before `Saving & Loading model`).
I had to replace the following line in the `MyDataset` class to avoid error of deprecated `item()` method from
`d = self.df.iloc[idx.item()]` to `d = self.df.iloc[idx]`. How do you think is such replacement correctly in this case?
After doing it I get insufficient results as you can see on the attachment. It's about 4 time worse result than you provide. 
Are there any ideas to improve results?

Torch: 1.7.0+cu110
Torchvision: 0.8.1+cu110
Pandas: 1.0.0
Ubuntu 18.04
![Screenshot from 2020-11-03 04-13-14](https://user-images.githubusercontent.com/17842321/97935923-08c94580-1d8b-11eb-93d3-e58478199033.png)

Thanks!
With best regards,
Andrey

**Comment by aman5319** [comment]
Regarding this `d = self.df.iloc[idx.item()] ` in pytorch 1.0.0 they use to pass torch tensor in the dataset `__getitem__ `method that's why item() was necessary to get the int value out of the tensor. I'm not sure whether in Pytorch 1.7.0 they pass torch tensor or plain old int value to index, upon looking at what you said it seems `__getitem__` method now takes int.

**Comment by aman5319** [comment]
How come you printing so many val loss and val acc in a list and no train loss and train Acc.
the code written by me is pretty basic and I don't think anything should change between Torch version 1.0.1  to 1.7.0 may be try debugging the code, or check whether you are using the pretrained model under **Model Definition** section or worse try installing torch 1.0.1 and run the code.

I tried running the exact code on google colab with only `item()` method modification mentioned by you.

![Screenshot from 2020-11-03 08-30-30](https://user-images.githubusercontent.com/19899771/97945933-fbbe4d80-1dae-11eb-87cc-a75ce810e1f0.png)


![Screenshot from 2020-11-03 08-30-40](https://user-images.githubusercontent.com/19899771/97945930-f9f48a00-1dae-11eb-8cbe-6ea6a0608ef3.png)

I got better results compared to the old one.
